packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1146,247 +1138,211 @@ describe("SharedTree", () => {
+					const rRemove = undoableRemoveOfOuterList(resubmitter);
+					const s1 = undoableInsertInInnerList(submitter, "s1");
+					const s2 = undoableInsertInInnerList(submitter, "s2");
 
-				provider.processMessages();
+					provider.processMessages();
 
-				assert.deepEqual([...resubmitTree.flexTree], []);
-				assert.deepEqual([...otherTree.flexTree], []);
+					submitter.assertOuterListEquals([]);
+					resubmitter.assertOuterListEquals([]);
+					const initialState = ["a", "s1", "s2", "r"];
+					submitter.assertInnerListEquals(initialState);
+					resubmitter.assertInnerListEquals(initialState);
 
-				performResubmitTest(provider, otherTree, resubmitTree, otherUndos, resubmitUndos);
-			});
+					resubmitter.setConnected(false);
 
-			it("two edits to a removed tree over the restore and edit of the removed tree", () => {
-				const provider = new TestTreeProviderLite(2);
-				const content = {
-					schema,
-					allowedSchemaModifications: AllowedUpdateType.None,
-					initialTree: [["a"]],
-				};
-				const resubmitTree = provider.trees[0].schematizeInternal(content);
-				const otherTree = provider.trees[1].schematizeInternal(content);
-				provider.processMessages();
+					assert.equal(s2.revert(), RevertibleResult.Success);
+					assert.equal(s1.revert(), RevertibleResult.Success);
+					submitter.assertOuterListEquals([]);
+					submitter.assertInnerListEquals(["a", "r"]);
 
-				const otherUndos = editTreeAndRemove(otherTree);
-				const resubmitUndos = makeTwoEdits(resubmitTree);
+					provider.processMessages();
 
-				provider.processMessages();
+					if (scenario === "restore and edit") {
+						assert.equal(rRemove.revert(), RevertibleResult.Success);
+						assert.equal(rEdit.revert(), RevertibleResult.Success);
+					} else {
+						assert.equal(rEdit.revert(), RevertibleResult.Success);
+						assert.equal(rRemove.revert(), RevertibleResult.Success);
+					}
+					resubmitter.assertOuterListEquals([["a", "s1", "s2"]]);
 
-				assert.deepEqual([...resubmitTree.flexTree], []);
-				assert.deepEqual([...otherTree.flexTree], []);
+					resubmitter.setConnected(true);
+					provider.processMessages();
 
-				performResubmitTest(
-					provider,
-					otherTree,
-					resubmitTree,
-					otherUndos,
-					resubmitUndos,
-					true,
-				);
-			});
+					const finalState = [["a"]];
+					submitter.assertOuterListEquals(finalState);
+					resubmitter.assertOuterListEquals(finalState);
+				});
+			}
 
-			it("two edits to a removed tree over the edit and restore of the removed tree", () => {
-				const provider = new TestTreeProviderLite(2);
-				const content = {
-					schema,
-					allowedSchemaModifications: AllowedUpdateType.None,
-					initialTree: [["a"]],
-				};
-				const resubmitTree = provider.trees[0].schematizeInternal(content);
-				const otherTree = provider.trees[1].schematizeInternal(content);
-				provider.processMessages();
+			for (const scenario of ["restore and edit", "edit and restore"] as const) {
```

Comment:
these for loops can be combined into one, I feel like the ordering of tests still makes sense